src/Core/Silk.NET.BuildTools/Generator.cs

```diff
@@ -217,9 +217,16 @@ private static void RunTask(BindTask task, Config config, Stopwatch? sw)
                         }
                     }
                     ClangConfig.SubstituteWindowsSdkPath(ref task);
+                    Profile coreProfile = null;
                     foreach (var src in task.Sources)
                     {
-                        profiles.Add(Clang.GenerateProfile(Path.GetFileName(src), File.OpenRead(GetPath(src)), task));
+                        var createdProfile = Clang.GenerateProfile(Path.GetFileName(src), File.OpenRead(GetPath(src)), task, coreProfile);
```

Comment:
Why do we need to know about the core profile here? Clang/ConvertConstruct should not make any linkage between profiles unrelated to each individual input/output - this is the bakery's job.

Reply:
The reason it is here is because the other profiles need to know which class name to use (the class name specifically is used for the T in NativeExtension<T>, which was wrong before this change)

Reply:
I thought we settled on something like `[Dawn:WebGPU]Dawn`

Reply:
> I thought we settled on something like `[Dawn:WebGPU]Dawn`

Well, it was like that, then i was told in this very request for pull, that the class name being wrong was a bug in the clang backend anyway, so i fixed it this way instead

Reply:
Yeah so the class name being wrong was a bug (thank you for fixing it!) - the class should always have the core class name, but when generating an extension we still need some way to pull it while remaining a input headers-output profile black box for the bakery.

Reply:
So should i have the class name patch and the `[Dawn:WebGPU]Dawn` syntax on top? I'm not sure how these would interact tbh

Reply:
Basically:
```
string className
string extClassName = get class name from opts.Clang.ClassName
if (opts.Clang.ClassName contains ":")
{
    className = get CORE class name from opts.Clang.ClassName
}
else
{
    className = extClassName // misnomer i know
}

// ...

new Class { Name = className }

// ...

new NativeApiSet { Name = extClassName }
```

Reply:
This has not been resolved.